src/PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Krivodonova.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Krivodonova.hpp"
+
+#include <cmath>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ScalarAdvection::Solutions {
+
+template <typename DataType>
+tuples::TaggedTuple<ScalarAdvection::Tags::U> Krivodonova::variables(
+    const tnsr::I<DataType, 1>& x, double t,
+    tmpl::list<ScalarAdvection::Tags::U> /*meta*/) const noexcept {
+  // map each grid points x(t) back to its initial position within [-1, 1] at
+  // t=0
+  auto x0 = make_with_value<tnsr::I<DataType, 1>>(x, 0.0);
+  get<0>(x0) = get<0>(x) - t;
+  for (size_t i = 0; i < get_size(get<0>(x)); ++i) {
+    auto& xi = get<0>(x0)[i];
+    xi = xi - 2.0 * floor(0.5 * (xi + 1.0));
+  }
+
+  // parameters for the initial profile (from the Krivodonova paper)
+  const double a{0.5};
+  const double z{-0.7};
+  const double delta{0.005};
+  const double alpha{10.0};
+  const double beta{log(2.0) / (36.0 * pow(delta, 2.0))};
+
+  const auto F = [](const double x_var, const double alpha_var,
+                    const double a_var) noexcept {
+    return sqrt(fmax(1.0 - pow(alpha_var, 2.0) * pow(x_var - a_var, 2.0), 0.0));
+  };
+  const auto G = [](const double x_var, const double beta_var,
+                    const double z_var) noexcept {
+    return exp(-beta_var * pow(x_var - z_var, 2.0));
+  };
+
+  // evaluate u(x,t) = u(x0,0)
+  auto u_variable = make_with_value<Scalar<DataType>>(x0, 0.0);
+  for (size_t i = 0; i < get_size(get<0>(x0)); ++i) {
+    const auto& xi = get<0>(x0)[i];
+    auto& ui = get(u_variable)[i];
+
+    if ((-0.8 <= xi) and (xi <= -0.6)) {
+      ui = (G(xi, beta, z - delta) + G(xi, beta, z + delta) +
+            4.0 * G(xi, beta, z)) /
+           6.0;
+    } else if ((-0.4 <= xi) and (xi <= -0.2)) {
+      ui = 1.0;
+    } else if ((0.0 <= xi) and (xi <= 0.2)) {
+      ui = 1.0 - abs(10.0 * xi - 1.0);
```

Comment:
on some systems/compilers `abs` operates only on integers (and sometimes silently!) giving the wrong answer. There are 3 ways of dealing with this:
1. if you know it's always `double` then `std::abs` will work
2. if you know it's always a `double` you can also do `fabs` (this is the standard C-way)
3. if you don't know what the type will be, you can add a line `using std::abs;` before this line stick with `abs`. This is the "most general" or "most correct" way in generic code.
Any of the 3 above are fine :)